lib/spack/spack/stage.py

```diff
@@ -823,7 +823,10 @@ def purge():
         for stage_dir in os.listdir(root):
             if stage_dir.startswith(stage_prefix) or stage_dir == '.lock':
                 stage_path = os.path.join(root, stage_dir)
-                remove_linked_tree(stage_path)
+                if os.path.isdir(stage_path):
+                    remove_linked_tree(stage_path)
+                else:
+                    os.remove(stage_path)
```

Comment:
Interestingly, this predates https://github.com/spack/spack/pull/27021, but if I understand correctly, `remove_linked_tree` now doesn't call `shutil.rmtree` with `ignore_errors=True`, so we have to actually consider the type of file we are looking at.

Reply:
Yes, that seems to be the case. The lock file was not removed, but since we said to ignore errors we never noticed, see https://github.com/spack/spack/issues/29687#issuecomment-1076676154